client/utils/guidance.py

```diff
@@ -136,42 +144,107 @@ def show_guidance_msg(self, unique_id, input_lines, access_token, hash_key, guid
-            return (self.misU_count_dict, self.tg_id, "")
+            # Check in answerDict to see if the student has ever given
+            # any of these wrong answers (sourced from dictWA2LstAssessNum_WA)
+            has_given_resp = any(other_resp in prev_responses for
+                                 _, other_resp in lst_assess_num)
+
+            if not has_given_resp:
+                log.info("Student has not given a response in dictWA2LstAssessNum_WA.")
+                print(GUIDANCE_DEFAULT_MSG)
+                return EMPTY_MISUCOUNT_TGID_PRNTEDMSG
+
+            # Check if the current wrong answer is in the question's dictWA2DictInfo
+            if not wa_details:
+                # Increment countDict by the number of wrong answers seen
+                # for each tag assoicated with this wrong answer
+                for misu in lst_mis_u:
+                    countData[misu] += min(len(prev_responses), 1)
+            else:
+                # Lookup the lst_mis_u of each wrong answer in the list of wrong
+                # answers related to the current wrong answer (lst_assess_num),
+                # using dictAssessNum2AssessId
+                assess_num_to_aid = self.guidance_json['dictAssessNum2AssessId']
+
+                # misu -> list of wrong answers for that
+                related_misu_tags_dict = {}
+
+                for related_num, related_resp in lst_assess_num:
+                    related_aid = assess_num_to_aid[related_num]
+                    # Get the lst_misu for this asssigmment
+                    related_info = self.guidance_json['dictAssessId2Info'].get(related_aid)
+                    if not related_info:
+                        log.info("Could not find related id: %s in info dict",
+                                 related_aid)
+                        continue
+                    related_wa_info = related_info['dictWA2DictInfo'].get(related_resp)
+                    if not related_info:
+                        log.info("Could not find response %s in %s info dict",
+                                 related_resp, related_aid)
+                        continue
+                    related_misu_list = related_wa_info['lstMisU']
+                    for misu in related_misu_list:
+                        existing_resps = related_misu_tags_dict.get(misu, [])
+                        # Add dictWA2DictInfo to list of responses for this misunderstanding.
+                        related_misu_tags_dict[misu] = existing_resps + [related_wa_info]
+
+                for misu, wa_info in related_misu_tags_dict.values():
+                    # Increment countDict for each tag in the set of tags for each related resp
+                    countData[misu] += 1
+
+                    # For each mis_u above threshold go through each related wrong answer
+                    # Add the msg_id that is given from lambda_info_misu to msg_id_set
+                    if countData[misu] > wa_count_threshold:
```

Comment:
This should not be needed if the loop `for mis_u in self.misU_count_dict:` is put back that I commented on above.